x-pack/plugins/security_solution/public/siem_migrations/rules/components/rule_details_flyout/index.tsx

```diff
@@ -73,38 +80,83 @@ interface MigrationRuleDetailsFlyoutProps {
 export const MigrationRuleDetailsFlyout: React.FC<MigrationRuleDetailsFlyoutProps> = React.memo(
   ({
     ruleActions,
-    ruleMigration,
+    migrationId,
+    migrationRuleId,
     matchedPrebuiltRule,
     size = 'm',
     extraTabs = [],
     closeFlyout,
   }: MigrationRuleDetailsFlyoutProps) => {
+    const { addError } = useAppToasts();
+
     const { expandedOverviewSections, toggleOverviewSection } = useOverviewTabSections();
 
-    const rule = useMemo(() => {
-      if (isMigrationCustomRule(ruleMigration.elastic_rule)) {
-        return convertMigrationCustomRuleToSecurityRulePayload(
-          ruleMigration.elastic_rule,
-          false
-        ) as RuleResponse; // TODO: we need to adjust RuleOverviewTab to allow partial RuleResponse as a parameter;
+    const { data: { ruleMigrations } = { ruleMigrations: [] }, isLoading: isDataLoading } =
+      useGetMigrationRules({
+        migrationId,
+        ids: [migrationRuleId],
+      });
```

Comment:
Why do we need to retrieve the migration data again, isn't it loaded in the table when we open the flyout? it would just be passed via props, right?

Reply:
Right, just overcomplicated things 😄  Will adjust it. As part of this changes, I exposed `ids` filter parameter, would you prefer to hide it?